opm/parser/eclipse/EclipseState/Tables/TableSchema.cpp

```diff
@@ -0,0 +1,59 @@
+/*
+  Copyright 2015 Statoil ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include <utility>
+
+#include <opm/parser/eclipse/EclipseState/Tables/TableSchema.hpp>
+
+namespace Opm {
+
+    TableSchema::TableSchema()
+    {
+
+    }
+
+    void TableSchema::addColumn(const ColumnSchema& column) {
+        m_columns.insert( column.name() , column );
+    }
+
+
+    const ColumnSchema&& TableSchema::getColumn( const std::string& name ) const {
+        return std::forward<const ColumnSchema>( m_columns.get_const( name ) );
+    }
+
+    const ColumnSchema&& TableSchema::getColumn( size_t columnIndex ) const {
+        return std::forward<const ColumnSchema>( m_columns.get_const( columnIndex) );
+    }
+
+    size_t TableSchema::size() const {
+        return m_columns.size();
+    }
+
+    /*std::vector<const ColumnSchema>::const_iterator begin() const {
+        return m_columns.begin();
+    }
+
+
+    std::vector<const ColumnSchema>::const_iterator end() const {
+        return m_columns.end();
+    }
+    */
```

Comment:
Thank you very much for all the comments, I am not in å position to address
them now - will tomorrow.
17. des. 2015 14:12 skrev "Andreas Lauser" <notifications@github.com>
følgende:

> In opm/parser/eclipse/EclipseState/Tables/TableSchema.cpp
> <https://github.com/OPM/opm-parser/pull/649#discussion_r47903380>:
>
> > +        return std::forward<const ColumnSchema>( m_columns.get_const( columnIndex) );
> > +    }
> > +
> > +    size_t TableSchema::size() const {
> > +        return m_columns.size();
> > +    }
> > +
> > +    /*std::vector<const ColumnSchema>::const_iterator begin() const {
> > +        return m_columns.begin();
> > +    }
> > +
> > +
> > +    std::vector<const ColumnSchema>::const_iterator end() const {
> > +        return m_columns.end();
> > +    }
> > +    */
>
> better remove instead of cluttering?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OPM/opm-parser/pull/649/files#r47903380>.
>